### PR TITLE
Add gemma4_vit_ architecture support to TIMM model prefixes

### PR DESCRIPTION
**Pull request by mrpositron**
## What has changed and why?

`test_architecture_info__all_timm_model_names_match_prefix` validates that every model returned by `timm.list_models()` matches a prefix in `_TIMM_ARCH_NAME_PREFIXES`. Since `gemma4_vit_` was missing, the test started failing.

## How has it been tested?

NA

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

**Comment by gabrielfruet** [comment]
Do you mind posting the error message?

File: src/lightly_train/_models/timm/timm.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
     ("sequencer2d_", {"model_type": "transformer", "norm_type": "layernorm"}),
     ("aimv2_", {"model_type": "transformer", "norm_type": "layernorm"}),
     ("convit_", {"model_type": "transformer", "norm_type": "layernorm"}),
+    ("gemma4_vit_", {"model_type": "transformer", "norm_type": "layernorm"}),
```
Comment (gabrielfruet) [inline src/lightly_train/_models/timm/timm.py:54]: https://github.com/huggingface/pytorch-image-models/blob/0e968e1cb8b18ea66e6f492ea9536ce21cf062b1/timm/models/gemma4_vit.py#L386-L389

They use RMSNorm in `gemma4_vit_*`

**Review by gabrielfruet — CHANGES_REQUESTED** [review]

File: src/lightly_train/_models/timm/timm.py (hunk 1 above)
Comment (mrpositron) [reply src/lightly_train/_models/timm/timm.py:54]: * The literal doesn't allow `rmsnorm`.  [src/lightly_train/_models/model_wrapper.py:46](https://github.com/lightly-ai/lightly-train/blob/fbcaeed5de254b7a08030d9fe68aee57ee4804c7/src/lightly_train/_models/model_wrapper.py#L42)

  * `norm_type: Literal["batchnorm", "layernorm"]` - afaik it's a binary tag, not a faithful description of the norm class.

* Existing entries cover families that exclusively or partially use RmsNorm and they're all tagged `layernorm`

Comment (gabrielfruet) [reply src/lightly_train/_models/timm/timm.py:54]: IDK, this abstraction doesn't seem to be used in any place. @liopeer is that okay?

Comment (mrpositron) [reply src/lightly_train/_models/timm/timm.py:54]: In my opinion, it is used to distinguish between training recipes for CNNs ("batchnorm") vs transformers ("layernorm").

Comment (gabrielfruet) [reply src/lightly_train/_models/timm/timm.py:54]: https://github.com/lightly-ai/lightly-train/blob/fbcaeed5de254b7a08030d9fe68aee57ee4804c7/src/lightly_train/_methods/distillationv3/distillationv3.py#L498-L518

This is the only branching that exists apparently. I would honestly rather have it specified to be an RMSNorm or add some kind of TODO just to be clear what is that way

**Review by liopeer — COMMENTED** [review]
LGTM!

**Review by gabrielfruet — APPROVED** [review]
